solr/solrj/src/java/org/apache/solr/common/cloud/DocRouter.java

```diff
@@ -38,14 +36,24 @@
  * @lucene.experimental
  */
 public abstract class DocRouter {
-  public static final String DEFAULT_NAME = CompositeIdRouter.NAME;
-  public static final DocRouter DEFAULT;
+  /**
+   * @deprecated Use {@link DocRouters#DEFAULT_NAME} instead.
+   */
+  @Deprecated(since = "9.8", forRemoval = true)
+  public static final String DEFAULT_NAME = DocRouters.DEFAULT_NAME;
```

Comment:
WDYT about keeping this as a constant string?  Then almost nobody would have a need to reference the new DocRouters.

Reply:
Keeping `getDocRouter(routerName)` would probably be another case of class initialization deadlock, so I moved forward and removed the deprecation and kept `DocRouters.DEFAULT_NAME`, so that it aligns with the class reference of `DocRouters.DEFAULT`.